hles/extraction/src/main/scala/org/broadinstitute/monster/dap/OkWrapper.scala

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.monster.dap
+
+import java.io.IOException
+import java.time.Duration
+
+import okhttp3.{Call, Callback, OkHttpClient, Request, Response}
+import org.broadinstitute.monster.common.msg.JsonParser
+import upack.Msg
+
+import scala.concurrent.{Future, Promise}
+
+class OkWrapper extends HttpWrapper {
+  /** Timeout to use for all requests to production RedCap. */
+  private val timeout = Duration.ofSeconds(60)
+
+  /** Construct a client instance backed by the production RedCap instance. */
+  val client = new OkHttpClient.Builder()
+    .connectTimeout(timeout)
+    .readTimeout(timeout)
+    .build()
+
+  def makeRequest(request: Request): Future[Msg] = {
+    val p = Promise[Msg]()
+    client
+      .newCall(request)
+      .enqueue(new Callback {
+        override def onFailure(call: Call, e: IOException): Unit =
+          p.failure(e)
+        override def onResponse(call: Call, response: Response): Unit = {
+          val maybeResult =
+            JsonParser.parseEncodedJsonReturningFailure(response.body().string())
+          maybeResult match {
+            case Right(result) => p.success(result)
+            case Left(err)     => p.failure(err)
+          }
+        }
+      })
+    p.future
+  }
+
+  override def toString = "OkHTTP wrapper class"
```

Comment:
❓ Why override `toString` here? Clearer logging?

Reply:
for some reason i was under the impression that it was a requirement to implement this for the `Serializable` trait, but presumably it can just use the default `Object.toString` implementation

Reply:
If this is Java's `Serializable` interface, it's only a marker interface/trait. I think the default `toString` should be ok.